### DISCUSSION TITLE
An error occured in "individual---save as template"

### DISCUSSION THREAD

**Discussion by Slow-Walker66** [Open-Systems-Pharmacology/Forum#1365] [Questions & Problems] [answered]
Hello, 

I was studying the project of dextromethorphan PBPK model released by SRuedesh (https://github.com/Clinical-Pharmacy-Saarland-University/Dextromethorphan-Model/tree/6a2630677ea3f52eb9ee4898677f61525c34e198), I found that when I reproduced it in PKsim 11 (the released file was built by PKsim 9), the predicted curve is lower than observed. So I tried to save the corresponding individual as the Template, however, when I uploaded, the error came as follows.

And I checked the file, I found the "CYP3A4|Human|Gorski 2004 PM, n=1" was right there, no missing.

Thank you in advance!

![image](https://user-images.githubusercontent.com/87760070/197949058-86458107-4add-4bb1-b664-cf882df9a94a.png)
![image](https://user-images.githubusercontent.com/87760070/197949288-fa1ffba9-22e4-45dc-902f-83964d84a6cd.png)

Application:
PK-Sim® 11.0.150

Key 'CYP3A4|Human|Gorski 2004 PM, n=1' could not be found

Stack trace:

at OSPSuite.Utility.Collections.Cache`2.<>c.<.ctor>b__10_0(TKey key)
   at OSPSuite.Utility.Collections.Cache`2.get_Item(TKey key)
   at PKSim.Presentation.Services.BuildingBlockTask.<>c__DisplayClass21_1.<SaveAsTemplate>b__2(IPKSimBuildingBlock reference)
   at OSPSuite.Utility.Extensions.EnumerableExtensions.Each[T](IEnumerable`1 list, Action`1 action)
   at PKSim.Presentation.Services.BuildingBlockTask.SaveAsTemplate(ICache`2 buildingBlocksWithReferenceToSave, TemplateDatabaseType templateDatabaseType)
   at PKSim.Presentation.Services.IndividualTask.SaveAsTemplate(IReadOnlyList`1 individuals, TemplateDatabaseType templateDatabaseType)
   at PKSim.Presentation.Services.BuildingBlockTask`1.SaveAsTemplate(IReadOnlyList`1 buildingBlocksToSave)
   at PKSim.Presentation.UICommands.SaveBuildingBlockAsTemplateCommand`1.PerformExecute()
   at OSPSuite.Presentation.UICommands.ObjectUICommand`1.Execute()
   at OSPSuite.Presentation.UICommands.UICommandExtensions.ExecuteWithinExceptionHandler(IUICommand uiCommand, IEventPub

**Comment by SRuedesh** [comment, marked as answer]
Hi there, 

there have been similar issues with this specific individual before when exporting/importing snapshots. I will try to look into this ASAP and get back to you.
As for the discrepancy in the simulation, I believe I know the source of the error. Somehow, PK-Sim tends to reset overwritten kcat values in compounds when exporting snapshots. Can you please make sure, that the kcat values (specifically for CYP3A4 and UGT2B15 for dextrorphan) are identical between these simulations? That should fix it.

Best regards, 
Simeon

**Reply by Yuri05** [reply]
@SRuedesh 
>Somehow, PK-Sim tends to reset overwritten kcat values in compounds when exporting snapshots.

I could not reproduce this problem. Can you please create a bug report for this (if not done yet)?

**Reply by SRuedesh** [reply]
@Yuri05 I have been able to reproduce it. However, this seems to only occur in the snapshot-export in v9.1 from the original file (tested 9.1 and 11). If you think this still is of interest, I am happy to open a separate issue with a reproducible example.

@Slow-Walker66 I have been able to reproduce the problem in the simulation and indeed the overwritten UGT2B15 kcat used for all simulations was reset during snapshot export. As for the individual, I think it was a naming issue. I will push a fixed version of the broken snapshots and let you know in the coming days.

**Reply by msevestre** [reply]
@SRuedesh Please yes let's create an issue

**Reply by Slow-Walker66** [reply]
> Hi there,
> 
> there have been similar issues with this specific individual before when exporting/importing snapshots. I will try to look into this ASAP and get back to you. As for the discrepancy in the simulation, I believe I know the source of the error. Somehow, PK-Sim tends to reset overwritten kcat values in compounds when exporting snapshots. Can you please make sure, that the kcat values (specifically for CYP3A4 and UGT2B15 for dextrorphan) are identical between these simulations? That should fix it.
> 
> Best regards, Simeon

Thank you a lot!
I checked it again, finding that the Kcat value of UGT2B15 and CYP3A4 had reset. It turned to the same curve after rewritting the value. 
<img width="650" alt="image" src="https://user-images.githubusercontent.com/87760070/198055265-ff43e787-bfed-4ec1-b82d-adc1141255df.png">
![image](https://user-images.githubusercontent.com/87760070/198055346-c3ee36f9-d1a7-4baa-b7c8-70f4065eede2.png)
